packages/cloud_firestore/lib/src/firestore.dart

```diff
@@ -52,4 +60,64 @@ class Firestore {
     assert(path != null);
     return new DocumentReference._(this, path.split('/'));
   }
+
+  /// Runs a set of atomic database operations.
```

Comment:
I think we need to document this a lot more, see https://firebase.google.com/docs/reference/ios/firebasefirestore/api/reference/Classes/FIRFirestore#/c:objc(cs)FIRFirestore(im)runTransactionWithBlock:completion: for an example

Include a mention of the default duration.

Reply:
Done!